src/datasources/blockchain/blockchain-api.manager.ts

```diff
@@ -0,0 +1,38 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { BlockchainApi } from '@/datasources/blockchain/blockchain-api.service';
+import { IBlockchainApi } from '@/domain/interfaces/blockchain-api.interface';
+import { IBlockchainApiManager } from '@/domain/interfaces/blockchain-api.manager.interface';
+import { IConfigApi } from '@/domain/interfaces/config-api.interface';
+import { Inject, Injectable } from '@nestjs/common';
+
+@Injectable()
+export class BlockchainApiManager implements IBlockchainApiManager {
+  private blockchainApiMap: Record<string, IBlockchainApi> = {};
+
+  constructor(
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+    @Inject(IConfigApi) private readonly configApi: IConfigApi,
+  ) {}
+
+  async getBlockchainApi(chainId: string): Promise<IBlockchainApi> {
+    const blockchainApi = this.blockchainApiMap[chainId];
+    if (blockchainApi) {
+      return blockchainApi;
+    }
+
+    const chain = await this.configApi.getChain(chainId);
+    this.blockchainApiMap[chainId] = new BlockchainApi(
+      this.configurationService,
+      chain,
+      this.destroyBlockchainApi,
+    );
+    return this.blockchainApiMap[chainId];
+  }
+
+  destroyBlockchainApi(chainId: string): void {
+    if (this.blockchainApiMap?.[chainId]) {
+      delete this.blockchainApiMap[chainId];
+    }
```

Comment:
This is great. In fact, being strict we should implement the same destroy mechanism in `transaction-api.manager.ts`, as the Transaction Service URLs can also change (even is it's pretty unusual). Now it requires a CGW restart to get a new configuration for a given `TransactionApi` datasource. What do you think?

Reply:
Agree that it's unusual but I think it would be a good pattern to copy. We could even consider creating a specific interface for these managers that we implement in both.

Reply:
I refactored the `TransactionApiManager` and `BalancesApiManager` to follow the same pattern with a common interface in https://github.com/safe-global/safe-client-gateway/pull/1640. When that is merged, I will propagate the changes into the `BlockchainApiManager` as well.